ipatests/test_integration/test_trust.py

```diff
@@ -993,3 +1043,168 @@ def test_server_option_with_unreachable_ad(self):
+            assert gid_str in test_group.stdout_text
+            assert grp_str in test_group.stdout_text
+            assert uid != gid
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_idoverride_with_auto_private_group(self, type):
+        """
+        Override ad trusted user in default trust view
+        and set auto-private-groups=[hybrid,true,false]
+        and ensure that overridden values takes effect.
+        """
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+
+        with self.set_idoverrideuser(nonposixuser,
+                                     self.uid_override,
+                                     self.gid_override
+                                     ):
+            self.mod_idrange_auto_private_group(type)
+            (uid, gid) = self.get_user_id(self.clients[0], nonposixuser)
+            assert uid == self.uid_override
+            assert gid == self.gid_override
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_nonposixuser_nondefault_primary_group(self, type):
+        """
+        Test for non default primary group.
+        For hybrid/false gid corresponds to the group testgroup1.
+        """
+        nonposixuser1 = "nonposixuser1@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(type)
+        (uid, gid) = self.get_user_id(self.clients[0], nonposixuser1)
+        if type == "true":
+            assert uid == gid
+        else:
+            test_group = self.clients[0].run_command(["id", nonposixuser1])
+            gid_str = "gid={0}(testgroup1@{1})".format(gid, self.ad_domain)
+            group = "groups={0}(testgroup1@{1})".format(gid, self.ad_domain)
+            assert (gid_str in test_group.stdout_text
+                    and group in test_group.stdout_text)
+
+
+class TestPosixAutoPrivateGroup(BaseTestTrust):
+    """
+    Tests for auto-private-groups option with posix AD trust
+    Related : https://pagure.io/freeipa/issue/8807
+    """
+    topology = 'line'
+    num_ad_domains = 1
+    num_clients = 1
+    num_ad_subdomains = 0
+    num_ad_treedomains = 0
+    uid_override = "99999999"
+    gid_override = "78878787"
+
+    def test_add_posix_trust(self):
+        tasks.configure_dns_for_trust(self.master, self.ad)
+        tasks.establish_trust_with_ad(
+            self.master, self.ad_domain,
+            extra_args=['--range-type', 'ipa-ad-trust-posix'])
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_gidnumber_not_corresponding_existing_group(self, type):
+        """
+        Test checks that sssd can resolve AD users which
+        contain posix attributes (uidNumber and gidNumber)
+        but there is no group with the corresponding gidNumber
+        and override ad trusted user in default trust view.
+        """
+        posixuser = "testuser2@%s" % self.ad_domain
+        with self.set_idoverrideuser(posixuser, uid="10060",
```

Comment:
Let me check with SSSD developers what is the expected behavior for a non-existing group (I am not sure whether we need to define the override). IMO it makes sense to test the behavior without the override but let's wait for SSSD's feedback.